Integrations/integration-Atlassian_Confluence_Server.yml

```diff
@@ -0,0 +1,650 @@
+  name: unsecure
+  defaultvalue: "true"
+  type: 8
+  required: false
+script:
+  script: |-
+    import requests
+    import json
+
+    requests.packages.urllib3.disable_warnings()
+
+    if not demisto.params()['proxy']:
+        del os.environ['HTTP_PROXY']
+        del os.environ['HTTPS_PROXY']
+        del os.environ['http_proxy']
+        del os.environ['https_proxy']
+
+    """
+    GLOBAL VARIABLES
+    """
+
+    SERVER = demisto.params()['url'][:-1] if demisto.params()['url'].endswith('/') else demisto.params()['url']
+    BASE_URL = SERVER + '/rest/api'
+    VERIFY_CERTIFICATE = not demisto.params().get('unsecure', False)
+    USERNAME = demisto.params()['username']['identifier']
+    PASSWORD = demisto.params()['username']['password']
+
+    """
+    Helper Functions
+    """
+    def return_error_and_exit(message):
+        demisto.results({
+            'Type': entryTypes['error'], # note, warning, error
+            'ContentsFormat': formats['text'], # json, markdown, file, text
+            'Contents': message
+        })
+        sys.exit(0)
+
+    """
+    Confluence Commands
+    """
+    def update_content(page_id, content_title, space_key, content_body, content_type, content_version):
+        content_data = {}
+        #Populate the content_data dictionary
+        content_data['type'] = content_type
+        if space_key is not None:
+            content_data['space'] = {"key": space_key}
+        if content_title is not None:
+            content_data['title'] = content_title
+        content_data['body'] = {
+            "storage": {
+                "value": content_body,
+                "representation":"storage"
+                }
+            }
+        content_data['version'] = {
+            "number": content_version
+            }
+
+        headers = {
+            'Content-Type': 'application/json',
+            'Accept': 'application/json'
+        }
+
+        full_url = BASE_URL + '/content/' + page_id
+
+        res = requests.put(full_url, verify=VERIFY_CERTIFICATE, auth=(USERNAME,PASSWORD), headers=headers, data=json.dumps(content_data))
+
+        if res.status_code < 200 or res.status_code >= 300:
+            return_error_and_exit('Failed to update Content.\nURL: {}, Status Code: {}, Response: {}'.format(full_url, res.status_code, res.text))
+
+        return res.json()
+
+    def update_content_command():
+        """
+        Confluence Update Content method
+        """
+
+        page_id = demisto.args().get('pageid')
+        content_title = demisto.args().get('title')
+        space_key = demisto.args().get('space')
+        content_body = demisto.args().get('body')
+        content_type = demisto.args().get('type')
+        content_version = int(demisto.args().get('currentversion')) + 1
+
+        raw_content = update_content(page_id, content_title, space_key, content_body, content_type, content_version)
+        content = {
+            "ID": raw_content['results']['id'],
```

Comment:
why don't you just use the values you passed to this function, would be more bulletproof

Reply:
Please explain.  I am not understanding this suggestion.